iina/PlayerWindowController.swift

```diff
@@ -0,0 +1,170 @@
+//
+//  PlayerWindowController.swift
+//  iina
+//
+//  Created by Yuze Jiang on 2/15/20.
+//  Copyright © 2020 lhc. All rights reserved.
+//
+
+import Foundation
+
+class PlayerWindowController: NSWindowController {
+  unowned var player: PlayerCore
+  
+  var menuActionHandler: MainMenuActionHandler!
+  
+  var loaded = false
+  
+  init(playerCore: PlayerCore) {
+    self.player = playerCore
+    super.init(window: nil)
+  }
+
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+  
+  internal lazy var followGlobalSeekTypeWhenAdjustSlider: Bool = Preference.bool(for: .followGlobalSeekTypeWhenAdjustSlider)
+  
+  /** Observers added to `UserDefauts.standard`. */
+  internal var notificationObservers: [NotificationCenter: [NSObjectProtocol]] = [:]
+  
+  @IBOutlet weak var volumeSlider: NSSlider!
+  @IBOutlet weak var muteButton: NSButton!
+  @IBOutlet weak var playButton: NSButton!
+  @IBOutlet weak var playSlider: NSSlider!
+  @IBOutlet weak var rightLabel: DurationDisplayTextField!
+  @IBOutlet weak var leftLabel: NSTextField!
+  
+  override func windowDidLoad() {
+    super.windowDidLoad()
+    loaded = true
+    
+    guard let window = window else { return }
```

Comment:
I would argue that this should crash.

Reply:
Then just use `!`?

Reply:
Yes.

Reply:
Maybe not in this PR, since there are a lot not in these files as well.